.github/workflows/prune-dist.yml

```diff
@@ -0,0 +1,39 @@
+# truncate the cockpit-dist history every Sunday night, to avoid unbounded growth
+name: prune-dist
+on:
+  schedule:
+    - cron: '0 1 * * 0'
+  # can be run manually on https://github.com/cockpit-project/cockpit/actions
+  workflow_dispatch:
+
+jobs:
+  run:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up configuration and secrets
+        run: |
+          printf '[user]\n\tname = Cockpit Project\n\temail=cockpituous@gmail.com\n' > ~/.gitconfig
+          echo '${{ secrets.COCKPITUOUS_TOKEN }}' > ~/.config/github-token
+          # we push to -dist repo via https://github.com, that needs our cockpituous token
+          git config --global credential.helper store
+          echo 'https://token:${{ secrets.COCKPITUOUS_TOKEN }}@github.com' >> ~/.git-credentials
+
+      - name: Clone -dist repo
+        run: |
+          git clone https://github.com/${{ github.repository }}-dist.git dist-repo
+
+      - name: Truncate history of -dist repo
+        run: |
+          set -ex
+          cd dist-repo
+          # oldest commit that applies to any still present tarball
+          REF=$(git log --pretty=format:%H * | tail -n1)
```

Comment:
I don't think that `*` is necessary there or is it?

Reply:
It is -- without it, you'd just get the *entire* log, and thus REF would always be the first-ever commit, and you'd not prune anything. The `*` expands to all currently "visible" tar files in the branch -- publish-dist already regularly `git rm`'s the ones older than 7 days. So if you do `git log 1234.tar | tail`, you'll get the oldest commit that introduced 1234; and if you do it for all tarballs with `*`, you'll get the oldest commit of all the current tarballs.